arangod/Aql/Query.cpp

```diff
@@ -149,6 +149,11 @@ Query::Query(std::shared_ptr<transaction::Context> const& ctx,
 
   _resourceMonitor.setMemoryLimit(_queryOptions.memoryLimit);
   _warnings.updateOptions(_queryOptions);
+
+  // store name of user that started the query
+  if (!ServerState::instance()->isDBServer()) {
+    _user = ExecContext::current().user();
+  }
```

Comment:
Is this your suggestion?
```suggestion
  // store name of user that started the query
  _user = ExecContext::current().user();
```